File: adventofcode-2016/src/test/java/nl/mvdr/adventofcode/adventofcode2016/day17/Part2Test.java
package nl.mvdr.adventofcode.adventofcode2016.day17;

import static org.junit.jupiter.api.Assertions.assertThrows;

import java.util.stream.Stream;

import org.junit.jupiter.api.Test;
import org.junit.jupiter.params.provider.Arguments;

import nl.mvdr.adventofcode.solver.SolverTest;

class Part2Test extends SolverTest<Part2> {
    
    Part2Test() {
        super(Part2.class);
    }
    
    static Stream<Arguments> testSolution() {
        return Stream.of(Arguments.of("370", "example-day17-1.txt"),
                Arguments.of("492", "example-day17-2.txt"),
                Arguments.of("830", "example-day17-3.txt"));
    }
    
    @Test
    void testNoValidPath() {
        var solver = new Part1();
        
        assertThrows(IllegalStateException.class, () -> solver.solve("example-day17-0.txt"));
    }
}

File: adventofcode-2025/src/test/kotlin/nl/mvdr/adventofcode/adventofcode2025/day09/Part2Test.kt
package nl.mvdr.adventofcode.adventofcode2025.day09

import nl.mvdr.adventofcode.solver.FunctionSolverTest
import org.junit.jupiter.params.provider.Arguments

class Part2Test: FunctionSolverTest<Long>(::solvePart1) {
    companion object {
        @JvmStatic
        fun testSolution(): List<Arguments> {
            return listOf(
                Arguments.of("24", "example-day09.txt")
            )
        }
    }
}